### DISCUSSION TITLE
Hudl 2021 update

### DISCUSSION THREAD

**Discussion by paulcarrm9mx** [remay/tmf-hudl#38] [Help]
Hi iam having issues with your software ive tried two methods 
* sd card with image file on it renamed image file update.img 
* rk batch tool

both fail 

sd card via the boot loader i scroll to rkimage update it flashes yellow with invalid image or invalid path.

rk batch tools once firmware ready to load i get a green square i click on the restore button the process starts for a few seconds the flashes a red bar with Successful {0) Failed{1}time 1379ms

the hudl is factory stock firmware

can you help please
![Capture55555](https://user-images.githubusercontent.com/93647513/140037094-6f51f896-2b8b-450f-bfdc-f98521ab1b24.PNG)

**Comment by remay** [comment]
Hi Paul,

The easiest method by far to get going is the SD Card method described here: https://github.com/remay/tmf-hudl/wiki/Flashing-from-an-SD-Card

[The RKBatchTool is somewhat tricky to get set up correctly, especially getting the right version of the drivers installed and mating the correct registry changes, and even then is somewhat dependent on the quality of the USB cable you have to hand]

For the SD method make sure you click on each step in the instruction page to expand it and see more detail.

There are several cases in this discussion board of people having similar problems to you.  The symptom you describe indicates that the update.img file is either not found (possible hardware problem with the SD card reader) or, more likely, isn't in the format expected which would mean one of the steps hasn't completed correctly.

I'm assuming that you're running Windows on your computer.  Let me know if you need more details of any of these steps:
1. Download the compressed image file by clicking here: https://github.com/remay/tmf-hudl/releases/download/tmf-custom-rom-v1.0/hudl.20140424.153851.TMF-Custom-v1.0.img.zip
2. Double click the downloaded `hudl.20140424.153851.TMF-Custom-v1.0.img.zip` file to open an explorer window that shows the content of the ZIP file - that should be a single file named `hudl.20140424.153851.TMF-Custom-v1.0.img`
3. Copy (drag & drop) `hudl.20140424.153851.TMF-Custom-v1.0.img` to the SD card
4. On the SD card rename `hudl.20140424.153851.TMF-Custom-v1.0.img` to `update.img`.

And then try following the steps you did before to use the SD card to flash the image and see if you get the same failure.   Let me know how you get on.

**Comment by paulcarrm9mx** [comment]
![image](https://user-images.githubusercontent.com/93647513/140041406-a1b7fee4-bb49-43f8-9547-294e3ce335bf.png)

like this

**Comment by paulcarrm9mx** [comment]
ok says can not found firmware image or invalid image. Update rkimage complete. but nothing happens

**Comment by paulcarrm9mx** [comment]
i have my tf card in a usb reader hence usb drive E:

**Comment by remay** [comment]
Just to check - did you insert the card into the Hudl before turning it on?  If not then retry with the card inserted before switching on.

It looks like you have the correct file on the micro SDHC card, but we'll come back and confirm that later.  (You refer to it as a TF card - my understanding is that TF is just an old name for micro SD)

Let's confirm your Hudl can read the SD card.
- Follow the same steps as before to get into recovery mode
- Select 'apply update from external storage'

If it can read the SD card then you should get a screen that starts 'Choose a package to install:'.  In this case select the '../' option (that may be the only option) and select (with the power button) to get back to the main recovery screen.

If it can't read the SD card then you'll end up back on the main recovery menu.

Let me know whether you think the device can read the card

**Reply by paulcarrm9mx** [reply]
Hi 

i get the following as instructed above

../
lost.dir/
system volume information/

when clicking update from external storage

**Reply by paulcarrm9mx** [reply]
> Just to check - did you insert the card into the Hudl before turning it on? If not then retry with the card inserted before switching on.
> 
> It looks like you have the correct file on the micro SDHC card, but we'll come back and confirm that later. (You refer to it as a TF card - my understanding is that TF is just an old name for micro SD)
> 
> Let's confirm your Hudl can read the SD card.
> 
> * Follow the same steps as before to get into recovery mode
> * Select 'apply update from external storage'
> 
> If it can read the SD card then you should get a screen that starts 'Choose a package to install:'. In this case select the '../' option (that may be the only option) and select (with the power button) to get back to the main recovery screen.
> 
> If it can't read the SD card then you'll end up back on the main recovery menu.
> 
> Let me know whether you think the device can read the card

tried with another micro sd card same issue doesnt like the file and cant fine the path

**Comment by remay** [comment]
OK.  So the card is good.  Let's try this:

1. Download `update.img` from https://we.tl/t-O1s5p0lVU7  (your virus scanner might not like it due to the `.img` extension, but it's fine, it's just the unzipped content of the release download)
2. Move `update.img` to be the only file on your SD card
3. Try the 'update rkimage from external storage' option again

Let me know if that works.

**Reply by paulcarrm9mx** [reply]
thank you that worked a treat.

**Reply by remay** [reply]
Would you be happy to help me try to work out what went wrong?   You're 4th or 5th person I've helped like this and I'd like to understand which of the steps isn't working so I can provide better instructions.

If you still have it then I'd like the update.img file that didn't work - if you could share it with me through WeTransfer?

**Reply by paulcarrm9mx** [reply]
Basically it's githubs file compression just upload it uncompressed. Although your technique worked the tablets hardware was passed using. 99% apps are no longer compatible with 4.4

**Reply by remay** [reply]
Sorry, I don't understand what you're saying.  What Github file compression are you talking about?  I'm expecting the end user to download a ZIP file and then extract/uncompress the enclosed img file.  It seems that when you (and others) do this you end up with a file that is somehow corrupted.   Is it possible for you to share the file that you had on your SD card when the flash process failed?

**Comment by paulcarrm9mx** [comment]
I will tomorrow iam off to work

**Comment by lampco** [comment]
remay - need some help hudl 1 will not load as per links attach is this hudl update 38
have tried all methods (even different SD cards) 
the  link-  https://we.tl/t-O1s5p0lVU7 -is no longer live-discontinued 
any help/advice appreciated 
mikel

**Reply by remay** [reply]
Hi @lampco - sorry for the Christmas induced delay.   Can we work together to see if we can work out where you're going wrong so that I can update the instructions.

If you have the same issue as on this thread then the problem is with downloading and extracting the image:  the link that has expired just accessed the un-zipped content of the [download from the releases page](https://github.com/remay/tmf-hudl/releases/download/tmf-custom-rom-v1.0/hudl.20140424.153851.TMF-Custom-v1.0.img.zip).

Let's try and use the SD card method.  Are you downloading and moving the image file to the SD card using a windows computer?  Are you happy for me to give you some instructions and other software to download so we can work out where you're having issues?

Thanks,
Rob.

**Comment by lampco** [comment]
Robert yes I would like to get it up & running - just for my satisfaction
. I am away from home until monday of next week & will check in with
details of what I have done to date
Regards
     Mike l

**Reply by remay** [reply]
OK, I'll wait for you to come back to this thread next week.

**Comment by Nucktheking** [comment]
Hi there: appreciate that this is an old thread, but I've been trying this for an hour or so now and hope you're still getting updates!

When I go to the system recovery menu, I don't have a "update rkimage from external storage".

I can choose "apply update from external storage" and it gives me the directory for the SD card: but there's only a "System Volume Information" folder, and when I select that, it's blank.

I have definitely followed the steps exactly: unzip the folder, copy the file to the root of the SD card; rename it; get to recovery menu (without starting the device: I made that mistake and it added some other folders to the SD card).

So my problem is that the "update rkimage from external storage" isn't an option.

Any ideas?

Thanks.

N.

**Reply by remay** [reply]
>  I don't have a "update rkimage from external storage"

Are you definitely attempting to update a Hudl 1.  There is no version of the Hudl 1 that I am aware of without  that option in the recovery menu.  Is your device a Hudl 2?     This image can only be used with a Hudl 1.

**Comment by Nucktheking** [comment]
Thanks so much for the reply; I had assumed that it was a Hudl 1, but having checked the model number, it is indeed a Hudl  2...!

Is there anything I can do with it?

N.

**Reply by remay** [reply]
You could try following the steps here: https://rob.themayfamily.me.uk/hudl/hudl-fix   I've had reports of it working with Hudl 2 devices (but I don't have one to check).  The warning about it not working at the top needs updating as I'm told something changed again and that the process does work for Hudl 1s again.

If you do try it I'd be interested in knowing whether you were successful or not.
